stock_no_negative/tests/test_stock_no_negative.py

```diff
@@ -166,16 +166,17 @@ def test_true_allow_negative_stock_product_with_lot(self):
         the allow_negative_stock is set active in the product with lot"""
         self.product_with_lot.allow_negative_stock = True
         self.stock_picking_with_lot.action_confirm()
-        self.stock_picking_with_lot.move_ids.quantity_done = 100
+        self.stock_picking_with_lot.move_ids.quantity = 100
         with self.assertRaises(UserError):
+            self.stock_picking_with_lot.move_ids.picked = True
             self.stock_picking_with_lot._action_done()
         self.stock_picking_with_lot.move_ids.move_line_ids[0].lot_id = self.lot1.id
         self.stock_picking_with_lot._action_done()
+        self.stock_picking_with_lot.button_validate()
         quant = self.env["stock.quant"].search(
             [
                 ("product_id", "=", self.product_with_lot.id),
                 ("location_id", "=", self.location_id.id),
-                ("lot_id", "=", self.lot1.id),
```

Comment:
because this change, this alter the purpose of the test

Reply:
> because this change, this alter the purpose of the test

Hi @celm1990, I have checked this same process in v17, 
In v16:
If I validate 5 qty in delivery 2 records will be generated in stock.quant, one with 5 other with -5 both have the lot number.

In v17:
If I validate 5 qty in delivery 2 records will be generated in stock.quant, one with 5 other with -5 but from these two only +5 one have the lot number.

Reply:
Hi @celm1990 , I updated the test case by looking into it more deeply, Detailed operation(move line) with lot should be created before the done qty updated in move if we want to negative stock with lot no.
So I updated as per that.
Can you please review this!

Reply:
Hi @celm1990 ,

I got your point, before I removed that because I created the move line(with lot) while the picking was created so the Usererror won't raise that's why I removed it. 

But I just updated that test case, and added the test case for that Usererror to be raised, and then I'm creating the Detaile operations (move line with lot).

Can you please review this!